File: backend/app/models/forensic.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any

class RegionBox(BaseModel):
    x: int
    y: int
    width: int
    height: int
    label: Optional[str] = "Suspicious Region"

class TamperingFinding(BaseModel):
    detector: str
    score: float  # 0.0 to 1.0 (higher = higher anomaly/risk)
    severity: str  # LOW, MEDIUM, HIGH
    status: str  # CLEAN, SUSPICIOUS, REVIEW_REQUIRED, UNABLE_TO_VERIFY
    region: Optional[RegionBox] = None
    explanation: str
    details: Dict[str, Any] = {}

class ForensicAnalysisResult(BaseModel):
    findings: List[TamperingFinding] = []
    overall_tampering_score: float = 0.0
    suspicious_regions_count: int = 0
    annotated_image_url: Optional[str] = None

File: frontend/src/types/forensic.ts
export interface RegionBox {
  x: number;
  y: number;
  width: number;
  height: number;
  label?: string;
}

export interface TamperingFinding {
  detector: string;
  score: number;
  severity: 'LOW' | 'MEDIUM' | 'HIGH';
  status: 'CLEAN' | 'SUSPICIOUS' | 'REVIEW_REQUIRED' | 'UNABLE_TO_VERIFY' | 'VERIFIED_PRESENT';
  region?: RegionBox;
  explanation: string;
  details: Record<string, any>;
}

export interface ForensicAnalysisResult {
  findings: TamperingFinding[];
  overall_tampering_score: number;
  suspicious_regions_count: number;
  annotated_image_url?: string;
}
